include/crypto/ecx.h

```diff
@@ -109,10 +109,19 @@ int X448(uint8_t out_shared_key[56], const uint8_t private_key[56],
 void X448_public_from_private(uint8_t out_public_value[56],
                               const uint8_t private_key[56]);
 
+
 /* Backend support */
 int ecx_public_from_private(ECX_KEY *key);
 int ecx_key_fromdata(ECX_KEY *ecx, const OSSL_PARAM params[],
                      int include_private);
 
+ECX_KEY *evp_pkey_get0_X25519(const EVP_PKEY *pkey);
```

Comment:
These seems like overkill. This is all we actually need..
```
ECX_KEY *evp_pkey_get1_ECX_KEY(EVP_PKEY *pkey, int type)
```

The encoder can do this
```
#define IMPLEMENT_ECX_NEWCTX(KEYTYPEstr, KEYTYPE, keytype, free)    \
static ECX_KEY *evp_pkey_get1_##keytype(EVP_PKEY *pkey)                        \
{                                                                              \
    return evp_pkey_get1_ECX_KEY(pkey, EVP_PKEY_##KEYTYPE);                    \
}                                                                              \
IMPLEMENT_NEWCTX(KEYTYPEstr, KEYTYPE, keytype, evp_pkey_get1_##keytype, free)
```

Reply:
Yeahok...  I was again thinking in "feature complete" terms, which is a bit silly for internal functions, I have to admit.

Reply:
Fixed